### ISSUE TITLE
when the angular-multi-select inside tab (tabset) output-module had Invalid

### ISSUE THREAD

**Issue by liulipengju** [isteven/angular-multi-select#178]
Hi isteven
     when the angular-multi-select inside tab (tabset) output-module had Invalid,see the following code
# code
`<form ng-submit="exportTables()">`
`	<div>`
`		<div>`
`			<label class="col-md-1">select table</label>`
`			<div class="col-md-6">`
`			<tabset style="margin-left:8px;">`
`				<tab heading="Table Manage"  >`
`					<div`
`						multi-select`
`						input-model="tableInit"`
`						output-model="tableOut"`
`						button-label="name"`
`						item-label="name maker"`
`						tick-property="ticked"`
`					>`
`					</div>	`			
`				</tab>`
`			</tabset>`
`			</div>`
`			<div class="col-md-4">`
`				<button type="submit" class="btn purple" ng-disabled="tableOut.length === 0">Submit</button>`
`			</div>`
`		</div>`
`	</div>`
`</form>`

# controller
$scope.tableInit = [];
        $scope.tableOut = [];
        $scope.tableItems = [
            {
                "tableName": "t_purchase",
                "tableDesc": "采购订单表"
            },
            {
                "tableName": "t_purchase_backgoods",
                "tableDesc": "采购退货表"
            }];

$scope.initSelectItems = function () {
            angular.forEach($scope.tableItems, function (q) {
                $scope.tableInit.push({
                    name: q.tableDesc,
                    maker: '',
                    ticked: false,
                    tableName: q.tableName
                });
            });
        };
$scope.exportTables = function () {
        	console.log($scope.tableOut);
};

you can try!

**Comment by liulipengju** [comment]
{
<form ng-submit="exportTables()">
	<div>
		<div>
			<label class="col-md-1">select table</label>
			<div class="col-md-6">
			<tabset style="margin-left:8px;">
				<tab heading="Table Manage"  >
					<div
						multi-select
						input-model="tableInit"
						output-model="tableOut"
						button-label="name"
						item-label="name maker"
						tick-property="ticked"
					>
					</div>				
				</tab>
			</tabset>
			</div>
			<div class="col-md-4">
				<button type="submit" class="btn purple" ng-disabled="tableOut.length === 0">Submit</button>
			</div>
		</div>
	</div>
</form>
}

**Comment by liulipengju** [comment]
> <form ng-submit="exportTables()">
	<div>
		<div>
			<label class="col-md-1">select table</label>
			<div class="col-md-6">
			<tabset style="margin-left:8px;">
				<tab heading="Table Manage"  >
					<div
						multi-select
						input-model="tableInit"
						output-model="tableOut"
						button-label="name"
						item-label="name maker"
						tick-property="ticked"
					>
					</div>				
				</tab>
			</tabset>
			</div>
			<div class="col-md-4">
				<button type="submit" class="btn purple" ng-disabled="tableOut.length === 0">Submit</button>
			</div>
		</div>
	</div>
</form>

**Comment by isteven** [comment]
Hi @liulipengju , 

Please replicate this problem in JsFiddle or Plunker and send me the URL.

**Comment by liulipengju** [comment]
Well, wait a minute, I handle it, give you a shared URL

**Comment by liulipengju** [comment]
Plunker tips: Unable to load streaming code. i can not share.

the code very easy.

<link href="//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css" rel="stylesheet">
  <link rel="stylesheet" href="angular-multi-select.css">
  
  <script src="//ajax.googleapis.com/ajax/libs/angularjs/1.2.16/angular.js"></script>
  <script src="//angular-ui.github.io/bootstrap/ui-bootstrap-tpls-0.12.0.js"></script>
  <script src="angular-multi-select.js"></script>
  <script src="app.js"></script>

page inside code just ok.

**Comment by isteven** [comment]
If Plunker doesn't work, try to use JsFiddle please. It would be much easier to see the code right away.

You don't need to stream, just save your code, and copy & paste the JsFiddle (or Plunker) URL back to me.

**Comment by liulipengju** [comment]
but JsFiddle  How to reference angular-multiselect

**Comment by isteven** [comment]
You can copy paste the whole 
- angular-multi-select.js script into the JS box.
- angular-multi-select.css into the CSS box.

Or, open the RAW file from Github and include these files on the left hand side menu in JsFiddle. Please read JsFiddle's documentation if you're still unsure.

**Comment by liulipengju** [comment]
http://jsfiddle.net/g744rhcw/1/

code in this 
but  reference lib not work

**Comment by isteven** [comment]
I didn't see the controller in your JsFiddle.

I also didn't see the ng-app initiation in your JsFiddle.

**Comment by liulipengju** [comment]
![qq 20150202151323](https://cloud.githubusercontent.com/assets/10510950/5996909/12f234d8-aaee-11e4-91ac-ca4507b40d30.jpg)

**Comment by liulipengju** [comment]
http://jsfiddle.net/g744rhcw/2/

**Comment by liulipengju** [comment]
left external resources is not load . why?

**Comment by liulipengju** [comment]
Still do not see it?

**Comment by isteven** [comment]
If you open your browser console, there are still some errors there. Look at those with red underline.

If you already define an AngularJs in the "external framework", you don't need to include it again in the "external resources"

![multiselect](https://cloud.githubusercontent.com/assets/5432100/5996961/062b2588-aaef-11e4-92e5-cf7d2d542e10.jpg)

I don't know why your external resources are not loaded. Have you initated them? 

I'm sorry I cannot help you on these, I only support my multi-select directive, not JsFiddle / Plunker / AngularJs.

**Comment by liulipengju** [comment]
my question just for angular-multi-select directive,
because angular-multi-select inside tab (tabset == bootstrap3) output-module had Invalid

**Comment by isteven** [comment]
Yes... 

But before you ask, please replicate the problem in JsFiddle or Plunker.

How can I help you if you gave me an invalid JsFiddler?

**Comment by liulipengju** [comment]
Yes, I was there to see JsFiddle a problem.

**Comment by isteven** [comment]
OK, basically.. please provide me with a JsFiddle or Plunker that is working & free of error. 

Only then I can help you with my directive.

**Comment by liulipengju** [comment]
Hi isteven

i fix error,now no error, but angular-multiselect  is not working.               
http://jsfiddle.net/g744rhcw/11/

**Comment by isteven** [comment]
Hi, 

There's still an error; 

![multiselect](https://cloud.githubusercontent.com/assets/5432100/5997681/c9b3c4ba-aaf9-11e4-80ae-2e35766efdcc.jpg)

**Comment by liulipengju** [comment]
yes . wait me one minute.

**Comment by liulipengju** [comment]
Hi isteven
Oh.  haha.        
http://jsfiddle.net/g744rhcw/14/  it's oK

please check.
thank you

**Comment by liulipengju** [comment]
you can copy  multiselect of div out tabset and run

**Comment by isteven** [comment]
OK it seems that, just like previously, there's a conflict with angular-ui. 

You have to choose only one, either my directive, or the angular-ui.

**Comment by liulipengju** [comment]
Hi, This problem can not solve it?

**Comment by liulipengju** [comment]
I can still use, but I have taken from the input items to the output items .

**Comment by isteven** [comment]
Oh.. ok it's like this:

You see, the tabset itself is a directive.. and.. there seems to be a problem using my directive within this directive. 

If you want to, just use Bootstrap's tab. Don't use the tabset.

**Comment by liulipengju** [comment]
Thanks, I will use bootstrap's tab.or  taken from the input items to the output items

**Comment by liulipengju** [comment]
Hi isteven
I do need to close this issue ?

**Comment by isteven** [comment]
Yes, please close the issue when you're done. Thank you.

**Comment by liulipengju** [comment]
yes, i will.
